nfsd/parse_test.go

```diff
@@ -0,0 +1,200 @@
+`,
+			stats: &nfsd.RPCStats{
+				ReplyCache: nfsd.ReplyCache{
+					Hits:    0,
+					Misses:  6,
+					NoCache: 18622,
+				},
+				FileHandles: nfsd.FileHandles{
+					Stale:        0,
+					TotalLookups: 0,
+					AnonLookups:  0,
+					DirNoCache:   0,
+					NoDirNoCache: 0,
+				},
+				InputOutput: nfsd.InputOutput{
+					Read:  157286400,
+					Write: 0,
+				},
+				Threads: nfsd.Threads{
+					Threads: 8,
+					FullCnt: 0,
+				},
+				ReadAheadCache: nfsd.ReadAheadCache{
+					CacheSize:      32,
+					CacheHistogram: []uint64{0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+					NotFound:       0,
+				},
+				Network: nfsd.Network{
+					NetCount:   18628,
+					UDPCount:   0,
+					TCPCount:   18628,
+					TCPConnect: 6,
+				},
+				RPC: nfsd.RPC{
+					RPCCount: 18628,
+					BadCnt:   0,
+					BadFmt:   0,
+					BadAuth:  0,
+					BadcInt:  0,
+				},
+				V2Stats: nfsd.V2Stats{
+					Null:     2,
+					GetAttr:  69,
+					SetAttr:  0,
+					Root:     0,
+					Lookup:   4410,
+					ReadLink: 0,
+					Read:     0,
+					WrCache:  0,
+					Write:    0,
+					Create:   0,
+					Remove:   0,
+					Rename:   0,
+					Link:     0,
+					SymLink:  0,
+					MkDir:    0,
+					RmDir:    0,
+					ReadDir:  99,
+					FsStat:   2,
+				},
+				V3Stats: nfsd.V3Stats{
+					Null:        2,
+					GetAttr:     112,
+					SetAttr:     0,
+					Lookup:      2719,
+					Access:      111,
+					ReadLink:    0,
+					Read:        0,
+					Write:       0,
+					Create:      0,
+					MkDir:       0,
+					SymLink:     0,
+					MkNod:       0,
+					Remove:      0,
+					RmDir:       0,
+					Rename:      0,
+					Link:        0,
+					ReadDir:     27,
+					ReadDirPlus: 216,
+					FsStat:      0,
+					FsInfo:      2,
+					PathConf:    1,
+					Commit:      0,
+				},
+				V4Stats: nfsd.V4Stats{
+					Null:     2,
+					Compound: 10853,
+				},
+				V4Ops: nfsd.V4Ops{
+					Op0Unused:    0,
+					Op1Unused:    0,
+					Op2Future:    0,
+					Access:       1098,
+					Close:        2,
+					Commit:       0,
+					Create:       0,
+					DelegPurge:   0,
+					DelegReturn:  0,
+					GetAttr:      8179,
+					GetFH:        5896,
+					Link:         0,
+					Lock:         0,
+					Lockt:        0,
+					Locku:        0,
+					Lookup:       5900,
+					LookupRoot:   0,
+					Nverify:      0,
+					Open:         2,
+					OpenAttr:     0,
+					OpenConfirm:  2,
+					OpenDgrd:     0,
+					PutFH:        9609,
+					PutPubFH:     0,
+					PutRootFH:    2,
+					Read:         150,
+					ReadDir:      1272,
+					ReadLink:     0,
+					Remove:       0,
+					Rename:       0,
+					Renew:        1236,
+					RestoreFH:    0,
+					SaveFH:       0,
+					SecInfo:      0,
+					SetAttr:      0,
+					Verify:       3,
+					Write:        3,
+					RelLockOwner: 0,
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		stats, err := nfsd.ParseRPCStats(strings.NewReader(tt.content))
```

Comment:
I'd do subtests: `t.Run(tt.name, func(t *testing.T) { /* ... */ })`

Reply:
@SuperQ The test error output currently doesn't show which of the test cases failed. What do you think about using subtests? You just need to wrap the whole block into the function @mdlayher pasted.

Reply:
Yes, sorry, got busy with other stuff, great idea.